File: lang/go/test/fun/vac/data_structures/ArrayStack/ArrayStackTest.go
package main

import (
    "fun/vac/data_structures/ArrayStack"
    "fun/vac/util/TestRunner"
)

func testArrayStack() bool {
    size := 8192

    stack := ArrayStack.New(0)

    for i := 1; i <= size; i++ {
        stack.Push(i)
    }

    stack.Shrink()

    if stack.Size() != size {
        return false
    }

    if stack.Capacity() != size {
        return false
    }

    for i := size; i > 0; i-- {
        if stack.Peek() != i {
            return false
        }
        stack.Pop()
    }

    stack.Shrink()

    if !stack.IsEmpty() {
        return false
    }

    if stack.Capacity() != 0 {
        return false
    }

    return true
}

func main() {
    TestRunner.ParseTest(testArrayStack())
}

File: lang/nodejs/test/vac.fun/data_structures/ArrayStackTest.js
"use strict"

const ArrayStack = require("vac.fun/data_structures/ArrayStack")
const TestRunner = require("vac.fun/util/TestRunner")

const testArrayStack = () => {
    let size = 8192

    const stack = new ArrayStack.ArrayStack(0)

    for (let i = 1; i <= size; i++) {
        stack.push(i)
    }

    stack.shrink()

    if (stack.size() !== size) {
        return false
    }

    if (stack.capacity() !== size) {
        return false
    }

    for (let i = size; i > 0; i--) {
        if (stack.peek() !== i) {
            return false
        }
        stack.pop()
    }

    stack.shrink()

    if (!stack.isEmpty()) {
        return false
    }

    return stack.capacity() === 0
}

if (module === require.main) {
    TestRunner.parseTest(testArrayStack())
}
